gnocchi/aggregates/rolling.py

```diff
@@ -0,0 +1,49 @@
+# -*- encoding: utf-8 -*-
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+import numpy as np
+from numpy.lib.stride_tricks import as_strided
+import six
+
+from gnocchi import aggregates
+
+
+def roll(values, window):
+    # http://arogozhnikov.github.io/2015/09/30/NumpyTipsAndTricks2.html
+    stride = values.strides[0]
+    return as_strided(values, shape=[len(values) - window + 1, window],
+                      strides=[stride, stride])
+
+
+def build_array(storage, metric, start, stop, granularity):
+    serie = storage.get_measures(metric, from_timestamp=start,
+                                 to_timestamp=stop, granularity=granularity)
+    ts = np.array(serie, dtype=[('timestamps', '<datetime64[ns]'),
```

Comment:
I'm feeling we could change `fetch` in Carbonara to directly return that format.